backend/src/test/java/com/elshop/backend/product/ProductUnitTest.java

```diff
@@ -0,0 +1,122 @@
+package com.elshop.backend.product;
+
+
+import com.elshop.backend.common.AppUtils;
+import com.elshop.backend.common.FakerUtils;
+import com.elshop.backend.exception.ResourceNotFoundException;
+import com.elshop.backend.product.model.Product;
+import com.elshop.backend.product.model.request.ProductRequest;
+import com.elshop.backend.product.model.response.ProductsListResponse;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+
+class ProductUnitTest {
+
+    private final ProductRepository mockRepository = mock(ProductRepository.class);
+    private final AppUtils utils = new AppUtils();
+    private final AppUtils mockUtils = mock(AppUtils.class);
+    private final ProductService productService = new ProductService(mockUtils, mockRepository);
+
+
+    @Test
+    void generateSlugTest() {
+        String title = "New 7aX 64GB";
+        String slug = utils.generateSlug(title);
+
+        assertTrue(slug.matches("(?i)^[a-z0-9-]+$"));
```

Comment:
Definiere Deine Erwartung so präzise und konkret wie möglich
```suggestion
        assertTrue(slug.matches("new-7ax-64gb-[a-z0-9-]{8}"));
```